### PR TITLE
mdspan support added

### PR DESCRIPTION
**Pull request by mpusz**
Specify library name and version:  **mdspan/0.1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mdspan/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                    "{} requires C++{} support. "
+                    "The current compiler {} {} does not support it.".format(
+                        self.name, self._minimum_cpp_standard,
+                        self.settings.compiler,
+                        self.settings.compiler.version))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "mdspan-mdspan-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
```
Comment (prince-chrismc) [inline recipes/mdspan/all/conanfile.py:54]: This is a new feature and it helps remove some of the recent bugs about `os.rename` not working on some platforms (it uses the internal `tools.rename` from conan which is more robust)

```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
One tiny comment and I can approve!

File: recipes/mdspan/all/conanfile.py (hunk 1 above)
Comment (mpusz) [reply recipes/mdspan/all/conanfile.py]: Thanks! Good to know :-D

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — DISMISSED** [review]
Perfect!

File: recipes/mdspan/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
```
Comment (SpaceIm) [inline recipes/mdspan/all/conanfile.py:5]:
```suggestion
import os

required_conan_version = ">=1.33.0"


```

due to `tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)`

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**